include/aws/s3/private/s3_meta_request_impl.h

```diff
@@ -264,6 +264,10 @@ void aws_s3_meta_request_set_fail_synced(
     struct aws_s3_request *failed_request,
     int error_code);
 
+/* Reset the request result. */
+AWS_S3_API
+void aws_s3_meta_request_reset_synced(struct aws_s3_meta_request *meta_request);
```

Comment:
Feel like the name could be better.  The comment says it resets the result but that's not a part of the name.  Also, document caller assumptions here and maybe add invariants that the lock is held when called.

Reply:
`aws_s3_meta_request_reset_synced` has "reset" in it. The assumption of held a lock when called is implied by "_synced", but yeah, we should document it out to make it clear